vividus-extension-selenium/src/main/java/org/vividus/monitor/GenericHarOnFailureMonitor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.monitor;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+import com.browserup.harreader.model.Har;
+import com.google.common.eventbus.EventBus;
+
+import org.vividus.context.RunContext;
+import org.vividus.proxy.har.HarManager;
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.util.json.JsonUtils;
+
+public class GenericHarOnFailureMonitor extends AbstractPublishingAttachmentOnFailureMonitor
+{
+    private static final String NO_HAR_ON_FAILURE_META_NAME = "noHarOnFailure";
+
+    private boolean harOnFailure;
+
+    private final HarManager harManager;
+    private final JsonUtils jsonUtils;
+
+    public GenericHarOnFailureMonitor(EventBus eventBus, JsonUtils jsonUtils, RunContext runContext,
+            IWebDriverProvider webDriverProvider, HarManager harManager)
+    {
+        super(runContext, webDriverProvider, eventBus, NO_HAR_ON_FAILURE_META_NAME);
+        this.jsonUtils = jsonUtils;
+        this.harManager = harManager;
+    }
+
+    @Override
+    protected void publishAttachment() throws IOException
+    {
+        takeAssertionFailureHar().ifPresent(har ->
+            attachPublishEvent(jsonUtils.toJsonAsBytes(har), "har.har"));
+    }
+
+    @Override
+    protected boolean isPublishingAttachmentOnFailureTake(Method method)
+    {
+        return !harOnFailure && method != null ? getPresentedAnnotation(method, PublishHarOnFailure.class) != null
+               : harOnFailure;
```

Comment:
```suggestion
        return harOnFailure || (method != null && getAnnotation(method, PublishHarOnFailure.class) != null);
```

Reply:
also keep in mind `method != null` won't be needed once `getAnnotation` is updated to incorporate this check